url8f84d8d956_klapaukh_JTuttePoly_tgz-pJ8-BitTutteJ8/src/Tutte.java

```diff
@@ -0,0 +1,349 @@
+import graph.Graph;
+import graph.SpanningGraph;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Scanner;
+
+import polynomial.FactorPoly;
+import polynomial.X;
+import polynomial.Y;
+import util.Debug;
+import util.Triple;
+
+public class Tutte {
+	MyTimer global_timer;
+	boolean status_flag;
+	boolean write_tree;
+
+	int resize_stats = 0;
+	long num_steps = 0;
+	long num_bicomps = 0;
+	long num_cycles = 0;
+	long num_disbicomps = 0;
+	long num_trees = 0;
+	long num_completed = 0;
+	long old_num_steps = 0;
+	int tree_id = 2;
+
+	boolean reduce_multicycles = true;
+	boolean reduce_multiedges = true;
+	boolean use_add_contract = false;
+	int split_threshold = 0;
+
+	public static void main(String[] args) {
+		if (args.length == 1) {
+			new Tutte(args[0]);
+		} else {
+			for (int i = 0; i < 100; i++) {
+				new Tutte(null);
+			}
+		}
+	}
+
+	public Tutte(String s) {
+		global_timer = new MyTimer();
+		num_steps = 0;
+		num_cycles = 0;
+
+		// Make a graph
+		int minVertex = 3;
+		int varVertex = 10;
+		int minEdges = 2;
+		int varEdges = 10;
+		int minMulti = 1;
+		int varMulti = 1;
+
+		SpanningGraph g = null;
+		long seed = 0;
+
+		if (s != null) {
+			try {
+				File f = new File(s);
+				Scanner scan = new Scanner(f);
+				scan.useDelimiter("[^0-9]+");
+				g = new SpanningGraph(30);
+				while(scan.hasNext()){
+//					System.out.println(scan.next());
+					
+					int from = scan.nextInt();
+					int to = scan.nextInt();
+					g.addEdge(from, to);
+				}
+//				return;
+			} catch (FileNotFoundException e) {
+				e.printStackTrace();
+			}
+		}
+		if (g == null) {
+			Random rand = new Random();
+			seed = rand.nextLong();
+			// seed =2001405263954154592L;
+			// System.out.println("Seed " + seed);
+			rand.setSeed(seed);
+
+			int numVertex = rand.nextInt(varVertex) + minVertex;
+			int numEdges = rand.nextInt(varEdges) + minEdges;
+
+			System.out.println("Graph with " + numVertex + " vertices and " + numEdges + " edges");
+			g = new SpanningGraph(numVertex);
+
+			for (int i = 0; i < numEdges; i++) {
+				int from = rand.nextInt(numVertex);
+				int to = rand.nextInt(numVertex);
+				int num = rand.nextInt(varMulti) + minMulti;
+				numEdges += num - 1;
+				Debug.debug("Adding " + from + " -> " + to + " x" + num);
+				g.addEdge(from, to, num);
+			}
+		}
+		FactorPoly tutte = tutte(g, 1);
+
+		System.out.println(tutte.toString());
+
+		// BigInteger correct = new BigInteger("2").pow(numEdges);
+		// BigInteger tuttei = tutte.substitute(2, 2);
+		// if (!correct.equals(tuttei)) {
+		// System.err.println("Correct T(2,2) = " + correct + " but we get " + tuttei);
+		// throw new RuntimeException("Sorry " + seed);
+		// }
+
+	}
+
+	private FactorPoly tutte(SpanningGraph graph, int mid) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 44 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.